facebook/ThreatExchange
# [hma][scripts] refactor helper script libs and add hma_shell cli

Summary
---------

The scripts are a ~hodgepodge of tools at the moment. This is an attempt at starting to standardize around `hma_script_utils.py` and clean up how the values needed to access HMA are passed to the testing scripts. 

It also added a prototype of a shell wrapper for HMA utils for interactive  testing.

Test Plan
---------

```
export HMA_USER_PWD='<redacted>' 
$ python3 scripts/hma_script_utils.py 
Manual Test of API Request Methods:
{'message': 'Hello World, HMA'} 

$ python3 scripts/hma_client_lib.py
This simple tests should take a little over 2 minutes to complete (due to sqs timeout).

Added configurations to HMA instance for test
Submitted content_id e2e-test-2021-08-17-d9a51f00-3d82-469d-a03c-6b6e4cb23e9e
Waiting for action history of submitted content_id
Checking every 5 seconds; maximum tries = 25
Success action event found in history!
Removed actions configurations used in test
Test completed in 161 seconds
```

hma_shell is still a work in program, will update docs with usage once functionality is expanded to a level that is useful.
Right now it can be used to run the same test as `hma_client_lib` and query the HMA API for its set of:
- action configs (actions and action_rules)
- dataset configs
- matches 

And can query for the following using a content_id:

- hash_details
- matches
- action_history

Usage:
```
$ python3 scripts/hma_shell 
Welcome! Type help or ? to list commands.

> ?

Documented commands (type help <topic>):
========================================
action_history_for_id  dataset_configs      help            refresh       
action_rules           exit                 matches         run_basic_test
actions                hash_details_for_id  matches_for_id
```
